### ISSUE TITLE
Custom Editor Click Behavior Issues

### ISSUE THREAD

**Issue by aviettran** [adazzle/react-data-grid#1429]
- Have you already searched for similar issues? Please help us out and double-check first!

- Also, please don't be that person who deletes this template. It's here for a reason.

- Thanks!

---

### Which version of React JS are you using?

✅ Officially supported ✅
- [ ] v15.4.x

⚠️ Not officially supported, expect warnings ⚠️
- [ ] v15.5.x
- [ ] v15.6.x

☣️ Not officially supported, expect warnings and errors ☣️
- [x] v16.x.x

---

### Which browser are you using?

✅ Officially supported ✅
- [ ] IE 9 / IE 10 / IE 11
- [ ] Edge
- [x] Chrome

⚠️ Not officially supported, but "should work" ⚠️
- [ ] Firefox
- [ ] Safari

---

### I'm submitting a ...

- [x] 🐛 Bug Report
- [ ] 💡 Feature Request

> 👋 Need general support? Not sure about how to use React itself, or how to get started with the Grid?
> Please do not submit support request here. Instead see

> https://github.com/adazzle/react-data-grid/blob/master/CONTRIBUTING.md

---

### Issue Details

```
Please include:
- What the current behavior is

1. Using a custom editor,
2. Edit a value
3. Click on a cell in the same column
4. The cell that is clicked on is updated with the updated value from the original cell edited

- What the desired behvaior is
The cell edited, not the cell clicked on, should be edited
- (If Bug) Steps to reproduce the issue
- (If Feature) The motivation / use case for the feature

We especially love screenshots / videos of problems, and remember
The Best Issue Is A Pull Request™
```
See:

https://codesandbox.io/s/30qkq8kkmm

**Comment by DominikHorn** [comment]
I have the exact same issue using react 16.6.3 and Chrome.

**Comment by PenguinOfThunder** [comment]
We are also seeing this issue with React 16.7.0 and Chrome.

**Comment by amanmahajan7** [comment]
@aviettran we just released an alpha version 6.0.2-0. Please give a shot and lets us know if everything is working as expected. Thanks

**Comment by DHorn** [comment]
My testing seems to indicate that the issue does not occur anymore. Thank you!
